### ISSUE TITLE
Epic: Take into account the AZ while connecting PS to SK

### ISSUE THREAD

**Issue by vadim2404** [neondatabase/neon#3200]
## Motivation
To reduce latency and expenditures on cross-AZ traffic, we need to consider the AZ while choosing the SK to which we need to connect. 

## DoD
The cross-AZ traffic has decreased.

## Implementation ideas
At first glance, we need to connect to SK the max LSN. But in our current situation, 2 of the 3 SKs will be with the same LSN. So, at the very beginning, I would suggest extending such criteria: when LSNs are equal, compare the SK AZ and PS SK and prefer to connect to SK in the same AZ.

## Tasks
- [x] start measuring cross-AZ traffic
- [ ] connect to closest SK in terms of the network when LSN difference is absent 

## Other related tasks and Epics
- neondatabase/cloud#2975

**Comment by vadim2404** [comment]
@neondatabase/safekeepers, guys, can you provide ETA for this task?

**Comment by petuhovskiy** [comment]
@vadim2404 I'm going to work on this, hope to add code for measuring cross-AZ traffic this week. If there will be no change in plans, it should reach staging before the end of next week.

**Comment by vadim2404** [comment]
Thanks for keeping me updated!

**Comment by petuhovskiy** [comment]
PR with metrics has reached prod, here is the query to look at traffic metrics 24 hours after deploy: https://neonprod.grafana.net/goto/K6NX5IfVz?orgId=1

Results look expected so far:
- 8% of safekeeper traffic is inter-AZ
- 16% of safekeeper traffic is cross-AZ
- 75% of safekeeper traffic has unknown source (but I'm almost sure it comes from computes)

We can also measure compute traffic, for that we only need to get access to AZ identifier inside a k8s pod and pass it in the connection string to safekeeper. That should fix unknown label on the 75% metrics but not urgent and can be done later.

16% of cross-AZ corresponds directly to PS connecting to SK from another AZ. That means that PR with advanced connection logic should move 16% of safekeeper traffic to inter-AZ category.

I also looked at AWS costs, at the `-DataTransfer-Regional-Bytes` which should be exactly cross-AZ traffic. I calculated we can only optimize 4% of our cross-AZ traffic costs with this epic, which is really not much compared to other EC2 costs.
